driver/src/main/java/org/neo4j/driver/internal/util/CertificateTool.java

```diff
@@ -147,6 +163,43 @@ public static void loadX509Cert( Certificate cert, String certAlias, KeyStore ke
     {
         keyStore.setCertificateEntry( certAlias, cert );
     }
+
+    private static X509Certificate generateSelfSignedCertificate(X500Name name, int bits, long validSecs)
```

Comment:
Not sure why we build this method. If it is to remove the warning while compiling mvn that `XX is internal proprietary API and may be removed in a future release`, then I think all the classes under `sun.security.x509` might not be supported in future versions. So this change will not help. But what we could do is to move this `genX509Cert` method into a test class as generating a cert is only used in tests. Then we might use `bouncycastle`